xonsh/history.py

```diff
@@ -407,103 +406,84 @@ def _hist_create_parser():
     return p
 
 
-def _hist_show(ns=None, hist=None, start_index=None, end_index=None,
-               start_time=None, end_time=None, location=None):
-    """Show the requested portion of shell history.
-    Accepts multiple history sources (xonsh, bash, zsh)
-
-    May be invoked as an alias with history all/bash/zsh which will
-    provide history as stdout or with __xonsh_history__.show()
-    which will return the history as a list with each item
-    in the tuple form (name, start_time, index).
+def _hist_get_session(session='session', location=None):
+    "Yield commands from history."
+    cmds = _HIST_SESSIONS[session](location=location)
+    try:
+        yield from cmds
+    except TypeError:
+        pass
+
+
+def _hist_get_portion(commands, slices):
+    "Yield from portions of history commands. "
+    commands = list(commands)
```

Comment:
i was planning on changing the session functions to generator funcs aswell(they can be refactored aswell anyway).  if we have only one slice (which may be the most common case) we can just islice the commands in `_hist_get_portion` . i will play a little more with islice.